sources/collections/tests/bit-vector-xor.dylan

```diff
@@ -40,72 +40,293 @@ define test bit-vector-xor-tiny-vector
   test-xor-for-all-pad-values("Xor two vectors again, arguments reversed",
-    vector6, vector5,
+    $huge-size, vector6, vector5,
     bits6-xor-bits5, bits6-xor-bits5, bits6-xor-bits5, bits6-xor-bits5);
 
 end test;
 
+define test bit-vector-xor-multiple-word-vector
+    (description: "Test bit-vector-andc2 with multiple-word sized bit-vectors")
+
+  let vector1 = make(<bit-vector>, size: $multiple-word-size);
+  let vector2 = make(<bit-vector>, size: $multiple-word-size);
+  let vector3 = make(<bit-vector>, size: $multiple-word-size, fill: 1);
+  let vector4 = make(<bit-vector>, size: $multiple-word-size, fill: 1);
+
+  let bits5 = random-elements($multiple-word-size, seed: 262);
+  let not-bits5 = compute-not-bits(bits5, $multiple-word-size);
+  let vector5 = make(<bit-vector>, size: $multiple-word-size);
+  set-bits(vector5, bits5);
+
+  let bits6 = random-elements($multiple-word-size, seed: 267);
+  let not-bits6 = compute-not-bits(bits6, $multiple-word-size);
+  let vector6 = make(<bit-vector>, size: $multiple-word-size, fill: 1);
+  unset-bits(vector6, not-bits6);
+
+  let bits5-xor-bits6
+    = intersection(union(bits5, bits6),
+                   compute-not-bits(intersection(bits5, bits6), $multiple-word-size));
+  let bits6-xor-bits5 = bits5-xor-bits6;
+
+  // Xor two all-zero vectors together with different pad values
+  //
+  test-xor-for-all-pad-values("Xor two all-zero vectors",
+    $multiple-word-size, vector1, vector2,
+    #"all-zeros", #"all-zeros", #"all-zeros", #"all-zeros");
+
+  // Xor two all-one vectors together with different pad values
+  //
+  test-xor-for-all-pad-values("Xor two all-one vectors",
+    $multiple-word-size, vector3, vector4,
+    #"all-zeros", #"all-zeros", #"all-zeros", #"all-zeros");
+
+  // Xor an all-zero and all-one vector together with different pad values
+  //
+  test-xor-for-all-pad-values("Xor an all-one and all-zero vector",
+    $multiple-word-size, vector1, vector3,
+    #"all-ones", #"all-ones", #"all-ones", #"all-ones");
+  test-xor-for-all-pad-values("Xor an all-zero and all-one vector",
+    $multiple-word-size, vector3, vector1,
+    #"all-ones", #"all-ones", #"all-ones", #"all-ones");
+
+  // Xor an all-zero vector with a mixed vector and different pad values
+  //
+  test-xor-for-all-pad-values("Xor an all-zero and mixed vector",
+    $multiple-word-size, vector1, vector5,
+    bits5, bits5, bits5, bits5);
+  test-xor-for-all-pad-values("Xor a mixed and all-zero vector",
+    $multiple-word-size, vector5, vector1,
+    bits5, bits5, bits5, bits5);
+
+  // Xor an all-one vector with a mixed vector and different pad values
+  //
+  test-xor-for-all-pad-values("Xor an all-one and mixed vector",
+    $multiple-word-size, vector3, vector5,
+      not-bits5, not-bits5, not-bits5, not-bits5);
+  test-xor-for-all-pad-values("Xor a mixed and all-one vector",
+    $multiple-word-size, vector5, vector3,
+      not-bits5, not-bits5, not-bits5, not-bits5);
```

Comment:
There are a few cases of inconsistent indentation like this.

(I suspect if I hit <tab> on most of these lines it will indent to underneath the first double quote.)

Reply:
emacs dylan-mode is a pain for re-indenting things when my attention is elsewhere.